app/src/main/java/com/geeksville/mesh/ui/SignalInfo.kt

```diff
@@ -27,7 +27,7 @@ fun signalInfo(
     } else {
         buildList {
             if (node.channel > 0) add("ch:${node.channel}")
-            if (node.hopsAway == 0) {
+            if (node.hopsAway == null) {
```

Comment:
So I'm assuming this is the key change and the rest above this is entirely stuff to make sure the DB stores nulls separate from zeros and that makes its way here for display. Let me know if that assumption is off!

I think we still want to show SNR & RSSI for nodes with both null and 0 hopsAway -- with null, it's sort of a best-effort equivalent to the current behavior; if the node's older, they might be meaningless, but that's just the risk. For actual 0, we definitely want it shown, since that's the case where the SNR/RSSI are definitely useful.

I think the summary is probably:

* display SNR/RSSI for both null and 0
* either display "Hops Away: 0" for those with an explicit 0 or display "Hops Away: unknown" for those with a null (or some other display that conveys this)
* make sure we're doing proper detection of the hopsAway/pulling it appropriately from nodedb, which I think is missing at present. In particular, in MeshService in processReceivedMeshPacket, the logic should be: if packet.hopStart == 0 or if hopLimit < hopStart (invalid), set hopsAway to null; else, set hopsAway to hopStart - hopLimit. I'm not sure where the code that instantiates this stuff on initial wantconfig lives in android, so that might be automatic there, but if not it should be doing something like (all pseudocode, heh): if !nodeInfo.hasHopsAway(): newNode.hopsAway = null; else: newNode.hopsAway = nodeInfo.hopsAway

Reply:
Gotcha - that helps, thank you!

Reply:
Ok @ianmcorvidae - I updated the UI logic to include rssi/snr values for `hopsAway == null || hopsAway == 0`, and am explicitly checking the proto `hasHopsAway` while converting an info packet to our local entity.

Let me know if I missed anything else - Andre will review as well for the rest of the infra/db changes.